cli/src/pcluster/validators/iam_validators.py

```diff
@@ -57,6 +59,33 @@ def _validate(self, policy: str):
             self._add_failure(str(e), FailureLevel.ERROR)
 
 
+class IamResourcePrefixValidator(Validator):
+    """
+    Iam Resource Prefix Validator.
+
+    Verify if the Resource Prefix belongs in the permission boundary if given.
+    """
+
+    IAM_PATH_NAME_REGEX = r"^((/[a-zA-Z0-9_.,+@=-]+)+)/"
+    IAM_ROLE_NAME_PREFIX = r"^[a-zA-Z0-9_.,+@=-]+$"
+
+    def _validate(self, resource_prefix: str):
+        try:
+            if not (
+                re.match(IamResourcePrefixValidator.IAM_PATH_NAME_REGEX, resource_prefix)
+                or re.match(IamResourcePrefixValidator.IAM_ROLE_NAME_PREFIX, resource_prefix)
+            ):
+                self._add_failure(
+                    f"Resource Prefix {resource_prefix} provided does not fall under the accepted pattern",
+                    FailureLevel.ERROR,
+                )
+        except Exception as e:
+            self._add_failure(
+                f"Resource Prefix {resource_prefix} provided does not fall under the accepted pattern",
```

Comment:
This error message is specifically meant for ResourcePrefix being in an invalid format.
If we have an exception here, it does not necessarily mean that. Right?

Reply:
I have removed the Exception section it was to handle one of the test cases where I had kept the resource prefix as None, however that condition is handled even before we call the Validation.